addons/mail/models/res_partner.py

```diff
@@ -126,14 +126,15 @@ def _message_fetch_failed(self):
         """Returns all messages, sent by the current partner, that have errors, in
         the format expected by the web client."""
         self.ensure_one()
-        messages = self.env['mail.message'].search([
-            ('has_error', '=', True),
+        notifications = self.env['mail.notification'].search([
             ('author_id', '=', self.id),
-            ('res_id', '!=', 0),
-            ('model', '!=', False),
-            ('message_type', '!=', 'user_notification')
+            ('notification_status', 'in', ('bounce', 'exception')),
+            ('mail_message_id.author_id', '=', self.id),
```

Comment:
Why adding a field on notification and then use the one on mail.message ? Don't understand ... especially there is already a filter on author right above.

Reply:
Well in theory the double condition on author is getting somewhat even more important without the `related` as both values could become unsync.

I have no idea if we would like to return messages from a different author if the notification has the right author or not... Adding the extra condition prevents it, removing the extra condition allows it. I guess we just have to decide.

Of course if we have to remove one, the main one to keep is the condition on notification as it's the one with the index.

Reply:
Well there is no flow allowing to unsync it, if the post-update if done (not sure it is done as ``_message_partner_info_from_emails`` is not updated). And in the end what matters is who authored the notification, unless I misunderstood something ?

Reply:
Yes it think it should be fine to consider only who authored the notification. Then we are conceptually splitting from a related, not just technically. Fine by my to make the change.